src/ApiGenerator/Domain/Specification/UrlInformation.cs

```diff
@@ -64,10 +65,23 @@ public IReadOnlyCollection<UrlPath> PathsWithDeprecations
 					.Where(grouped => !canonicalPartNameLookup.Any(set => set.SetEquals(grouped.Key)))
 					.Select(grouped => grouped.First().deprecatedPath);
 
-
 				_pathsWithDeprecation = paths
 					.Concat(withoutDeprecatedAliases.Select(p => new UrlPath(p, OriginalParts, Paths)))
 					.ToList();
+
+				// now, check for and prefer deprecated URLs
+
+				foreach (var path in _pathsWithDeprecation.Where(p => p.Deprecation is null).ToArray())
```

Comment:
LGTM but it would be better if we can do these mutations in a temporary variable and assign to `_pathsWithDeprecation` after the `foreach`

Reply:
Yep, good shout, updated.